src/linux/mem_reader.rs

```diff
@@ -105,26 +109,84 @@ impl MemReader {
         Ok(output)
     }
 
-    pub fn read(&mut self, src: usize, dst: &mut [u8]) -> Result<usize, CopyFromProcessError> {
+    pub fn read_pod<T: Plain>(&mut self, address: usize) -> io::Result<T> {
+        fn as_bytes_mut<T>(obj: &mut T) -> &mut [u8] {
+            unsafe {
+                std::slice::from_raw_parts_mut(obj as *mut _ as *mut u8, std::mem::size_of::<T>())
+            }
+        }
+        // Safety: All of this is safe to do because `Plain` is an unsafe trait that may only be
+        // implemented on types that are valid for every possible bit pattern, so there is nothing
+        // that we could read from the other process that isn't a valid value for our type.
+        let mut pod_obj: T = unsafe { std::mem::zeroed() };
+        let bytes = as_bytes_mut(&mut pod_obj);
+        self.read_exact(address, bytes)?;
+        Ok(pod_obj)
+    }
+
+    pub fn read_pod_vec<T: Plain>(
+        &mut self,
+        mut address: usize,
+        count: usize,
+    ) -> io::Result<Vec<T>> {
+        let mut v = Vec::with_capacity(count);
+        for _ in 0..count {
+            v.push(self.read_pod(address)?);
+            address += std::mem::size_of::<T>();
+        }
+        Ok(v)
+    }
+
+    pub fn read_until(
+        &mut self,
+        mut address: usize,
+        byte: u8,
```

Comment:
nit: Let's call this argument `terminator` to make the intent clearer

Reply:
I changed it. I had actually mostly named it this to be inline with the [`std` trait function of the same name](https://doc.rust-lang.org/stable/std/io/trait.BufRead.html#method.read_until)